integration-box/pelion-device-management/pelion_device.py

```diff
@@ -0,0 +1,75 @@
+import os
+import sys
+os.system(f"{sys.executable} -m pip install mbed-cloud-sdk")
+os.system(f"{sys.executable} -m pip install -U pytd==1.0.0 td-client")
+
+
+def _cast(value):
+    try:
+        value = int(value)
+    except ValueError:
+        try:
+            value = float(value)
+        except ValueError:
+            value = str(value)
+    return value
+
+
+def pull_resource_values(device_id, database, table):
+    import pytd
+    import pandas as pd
+    from mbed_cloud import ConnectAPI
+    from prestodb.exceptions import PrestoUserError
+
+    # see PDM Python client ConnectAPI examples:
+    # https://www.pelion.com/docs/device-management/current/mbed-cloud-sdk-python/subscriptions.html
+    # https://github.com/ARMmbed/mbed-cloud-sdk-python/blob/c2bc539856cc6932e367ed47f7c2e64ef6e3a77a/examples/connect/notifications.py#L25-L45
+    api = ConnectAPI({
+        'api_key': os.environ.get('PDM_API_KEY'),
+        'host': os.environ.get('PDM_HOST')
+    })
+
+    # check device existence
+    try:
+        filters = {'id': {'$eq': device_id}}
+        api.list_connected_devices(filters=filters).data
+    except Exception:
+        raise ValueError('cannot find the device: {}'.format(device_id))
+
+    column_names, row = [], []
+    for r in api.list_resources(device_id):
+        # observable resources whose values can change over time:
+        # https://www.pelion.com/docs/device-management/current/connecting/collecting-resources.html
+        if r.type is None or not r.observable:
+            continue
+
+        try:
+            value = api.get_resource_value(device_id, r.path)
+        except Exception as e:
+            print('skipped resource path {} due to {}'.format(r.path, e))
+            continue
+        value = _cast(value)
+        row.append(value)
+
+        column_names.append(r.type)
+
+        print('path: {}, type: {}, value: {}'.format(r.path, r.type, value))
+
+    if len(row) == 0:
+        sys.exit("no resource values from device: {}".format(device_id))
+
+    client = pytd.Client(apikey=os.environ.get('TD_API_KEY'),
+                         endpoint=os.environ.get('TD_ENDPOINT'),
+                         database=database)
+    df = pd.DataFrame(**{'columns': column_names, 'data': [row]})
```

Comment:
Just confirmation. How many records do you assume? Is it OK to upload in a batch from the memory consumption perspective?

Reply:
Single record per execution. 

There are some other options e.g., looping in the Python code and uploading multiple records at once, but I'd like to keep it straightforward for now; extra loop makes running time of this script longer, and long-running script is likely to be an unexpected use of the Custom Scripting capability..

Reply:
Make sense. That's why you're using insert_into :)